duckduckgo/Android
# Issue #753 - Downloads that open in a new tab leave user on an empty tab

Task/Issue URL: https://github.com/duckduckgo/Android/issues/753

**Description**:
* **Actual**: When downloading via a link that has target="_blank", the user is left with a blank tab.
* **Expected**: Ensure that this action does not open a new blank tab.
* Referenced the work of @AndrewR8 and the comments in PR https://github.com/duckduckgo/Android/pull/798

**Approach**:
1. Modified TabEntity to maintain a sourceTabId to reference the source tab by its tab id.
2. When opening a new tab, store the sourceTabId in the TabEntity db entry.
3. In the event that a download is opened in a new, blank tab (url is blank), callbacks will use tabRepo to select the source tab and delete the blank tab.

**Concerns**
* This still opens a blank tab in which the download options are displayed, but upon the user completing their selections, the blank tab is cleaned up and the source tab is selected and displayed.
* TabEntities now maintain a source whether it is needed or not - Could simplify this a bit by just passing the sourceTabId in the Fragment bundle as well?
* LeakCanary is identifying a leak, but struggling to track it down. If this approach is valid, could use some assistance on this.


**Steps to test this PR**:
1. Navigate to : https://www.w3schools.com/code/tryit.asp?filename=GGETHXPDSEMN
2. Run and select "Download in new tab".
3. Make the appropriate selection
4. New, blank tab should close and return to source tab.

---
![753](https://user-images.githubusercontent.com/2373024/86518482-1d5af880-be00-11ea-9e7a-fc5f8f411f68.gif)
